controllers/database/postgresql.go

```diff
@@ -1,36 +1,75 @@
 package database
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/go-logr/logr"
 	goharborv1 "github.com/goharbor/harbor-cluster-operator/api/v1"
+	"github.com/goharbor/harbor-cluster-operator/controllers/k8s"
 	"github.com/goharbor/harbor-cluster-operator/lcm"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
 )
 
 type PostgreSQLReconciler struct {
```

Comment:
A quick overview, it seems too many references are kept in the Reconciler, are they all needed to be kept as struct properties?